app/models/notification.rb

```diff
@@ -13,6 +13,12 @@ class Notification < ActiveRecord::Base
   scope :visible , lambda { joins('LEFT JOIN topics ON notifications.topic_id = topics.id')
     .where('topics.id IS NULL OR topics.deleted_at IS NULL') }
 
+  scope :get_liked_by, ->(user) {
+    where("data::json ->> 'original_username' = ?", user.username_lower)
```

Comment:
Yup this can be slow because we need to cast to `json` in Postgres. Technically, `Notification#data` needs to be migrated to use a `jsonb` column type but that is going to be a whole can of worms. The consideration I made here is that the use case restricts this query to `liked` notifications that have been made within the last 2 hours. Therefore, we'll have to cast the string to json for a limited set of records.